vpr/src/base/SetupGrid.cpp

```diff
@@ -183,10 +195,18 @@ static DeviceGrid auto_size_device_grid(const std::vector<t_grid_def>& grid_layo
 
             limiting_resources = grid_overused_resources(grid, minimum_instance_counts);
 
+            //Determine grid size
+            grid_size = width * height;
+
             //Increase the grid size
             width++;
 
-        } while (true);
+        } while (grid_size < max_size);
+
+        //Maximum device size reached
+        VPR_FATAL_ERROR(VPR_ERROR_OTHER,
+                        "Device auto-fit aborted: device size already exceeds required resources count by 100 times yet still cannot fit the design. "
+                        "Might be using more instances of a particular type of resource than the StratixIV devices can support (e.g. PLLs)\n");
```

Comment:
Should make this 100 (really 10,000 now) a defined constant so the message and code are guaranteed to agree.
Should make this message a bit more generic -- "This may be due to resources that do not grow as the grid size increases (e.g. PLLs in the Titan Stratix IV architecture capture."